### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by brandonandzeus** [Lilker712/MonsterTrainMod.NewCards#3]
1. SpellweaverSirenUnit

I take it the effects are Resolve: Generate a random spell.

But it seems from your code you'd want the random spell to be any spell card in the MegaPool. That's generally a bad idea.
When a run starts only the cards within the clan combination are loaded.

So if you were playing Hellhorned / Awoken and it pulls "Making of a Morsel" (an Umbra card) the game won't load the card and will pop up an error message saying that the "Game data is corrupted. Cards and characters may not display their art assets correctly."

From a design standpoint, its not useful to get a random card out of all of the cards in the game. Since, you could also get any card from the Wurmkin clan which is dependent on their Charged Echoes mechanic. It would be completely useless if you aren't playing Wurmkin.

Probably more workable would be to generate a random spell within the current clan combination 
use FilterBasedOnMainSubClass = true 

```
            var TrigEffect0 = new CardEffectDataBuilder
            {
                EffectStateType = typeof(CardEffectAddBattleCard),
                ParamInt = (int)CardPile.HandPile,
                ParamCardPool = CustomCardPoolManager.GetMegaPool(),
                ParamCardUpgradeDataBuilder = UpgradeEffect,
                FilterBasedOnMainSubClass = true,
                ParamCardFilter = new CardUpgradeMaskDataBuilder
                {
                    CardUpgradeMaskID = "OnlySpells",
                    CardType = CardType.Spell,
                }.Build(),
            };
```

2.  "IceCrystal"

The issue with this is that you are using the Reserve trigger which an effect that touches units other than the Pyre (The "apply [frostbite] [effect1.status0.power] to enemy units" part)

Note that combat previews don't take into account the Reserve trigger, so the combat preview won't account for the card applying Frostbite to the enemy units. The other stuff such as the pyre damage and freezing a card is okay.

**Comment by Lilker712** [comment]
Thanks a lot for helping me with "SpellweaverSirenUnit"
Now it is working correctly, I agree that I shouldn't get all spells from the game.
I needed to update the "TrainworksModdingTools" file maybe because of that my unit wasn't working correctly.

About "IceCrystal" from my perspective is working enough correctly, but still thanks for pointing it out.
I edit a little of his description.

**Comment by brandonandzeus** [comment]
Yes it works, but its not a good user experience to do something to have the combat previews be incorrect if you can't help it. 

For instance, with the Living Armor fight (ring 7 boss who has Revenge: +1 Attack) having Frostbite trigger once could vastly change the combat result.

The devs never intended for effects (other than healing/damaging) the pyre be present in that CardTrigger [see cards that have an OnPlayed trigger](https://github.com/search?q=repo%3Abrandonandzeus%2FMonsterTrainGameData%20OnUnplayed&type=code), so that's why it's not included in the preview math.

Just my two cents, and informing you on how that works.

**Comment by Lilker712** [comment]
I agree, I didn't think about this situation, So I decided to change the "IceCrystal" effect.
Thanks for the feedback. :)
